joyent/node-krill#4: add support for outputting LDAP search filter

misterdjules opened the pull request:
This is a work in progress. If that looks good, then I imagine we'll want to bump at least the minor version number in `package.json` at some point.

davepacheco commented on lib/krill.js:
```diff
@@ -470,6 +524,28 @@ function krillPrimValidateFieldType(fieldtypes, pred, key)
 
 }
 
+function sanityCheck(pred) {
```
I think it would be useful to document this a little.  Just reading through the file, it's not clear when this is called or what it's checking.

misterdjules replied:
Absolutely, sorry for not doing that, will do in the next update.

davepacheco commented on package.json:
```diff
@@ -1,20 +1,21 @@
-	    "jsprim": "0.5.1",
-	    "verror": "1.3.6"
-	}
+  "name": "krill",
```
Is there a reason to change the indenting on this file?

misterdjules replied:
This is the result of running `npm install --save assert-plus` with npm 1.4.28. I don't know why the indentation changed, but I thought that might be because the previous file was generated by an older (major?) version of npm, and that we'd want to keep the style generated by a more "current" version of npm.

This may not be what happened, so that's a great question.

davepacheco commented on lib/krill.js:
```diff
@@ -201,39 +210,84 @@ PredicateStream.prototype.stats = function ()
-    'eq': new RelationalOperator(
-	'==', [ 'number', 'string', 'boolean' ],
+    'le': new RelationalOperator({
+		names: {
```
There are a few undocumented style conventions that I've been using: keys used in objects are quoted (like 'le' is here), and nesting for inline objects like this is a continuation (i.e., 4-space indents).  Would you mind updating this?  Sorry that's not documented or checked, but I'd like to keep the repo consistent.

misterdjules replied:
Acknowledged, will do!

misterdjules commented:
@davepacheco I pushed a new separate commit, hopefully that addresses your comments. I'm not sure if I got the indentation of nest object literals right though. It would be great to have a way to automatically check that style to avoid going back and forth.

davepacheco commented:
Looks good!  Thanks.  Assuming it's `make prepush` clean, feel free to integrate and publish.  (I can also take care of that.)

misterdjules commented:
Merged with 1252d0b0012ae96e24249e21074a3947afb4ae3c, thank you!
